### PR TITLE
Refactored decode position and made position test

### PR DESCRIPTION
**Pull request by IonLi09**
We added test_position.py

### CODE REVIEW

**Review by David-Rey — DISMISSED** [review]
lgtm

File: tests/format/test_position.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# import EosLib
```
Comment (DarylDohner) [inline tests/format/test_position.py:1]: remove unnecessary line

File: EosLib/format/position.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def set_validity(self):
             self.valid = True
 
     @staticmethod
-    def decode_position(gps_packet: Packet):
+    def decode_position(gps_packet: Packet | bytes):
         new_position = Position()
 
         if isinstance(gps_packet, Packet):
```
Comment (DarylDohner) [inline EosLib/format/position.py:52]: Line 53 is wrong, position has no property called gps_packet
I think you just want `gps_packet.data_header.data_type`

File: tests/format/test_position.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+from datetime import datetime
+from EosLib.format.position import Position, FlightState
+
+
+
+def test_position():
```
Comment (DarylDohner) [inline tests/format/test_position.py:8]: you also need to test the flow where decode_position takes a Packet as the parameter

**Review by DarylDohner — CHANGES_REQUESTED** [review]
See comments.  You also need to update setup.py to bump the version to 2.0.1

File: tests/format/test_position.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    decoded_new_data = Position.decode_position(packet)
+
+    assert current_time == decoded_new_data.timestamp
+    assert 23.4 == decoded_new_data.latitude
+    assert 23.4 == decoded_new_data.longitude
+    assert 23.4 == decoded_new_data.altitude
+    assert 23.4 == decoded_new_data.speed
+    assert 5 == decoded_new_data.number_of_satellites
+    assert FlightState.NOT_SET == decoded_new_data.flight_state
+    assert decoded_new_data.valid
```
Comment (DarylDohner) [inline tests/format/test_position.py:58]: need newline at end of file

File: tests/format/test_position.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    # new_data.set_validity()
+    time = current_time
+    encoded_new_data = new_data.encode_position(time.timestamp(), 23.4, 23.4, 23.4, 23.4, 5,
+                                                FlightState.NOT_SET)
+
+    decoded_new_data = Position.decode_position(encoded_new_data)
+
+    assert time == decoded_new_data.timestamp
+    assert 23.4 == decoded_new_data.latitude
```
Comment (DarylDohner) [inline tests/format/test_position.py:28]: make all the numbers different, if you use the same numbers then you wouldn't know if they got accidentally switched somehow

File: tests/format/test_position.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    assert 23.4 == decoded_new_data.longitude
+    assert 23.4 == decoded_new_data.altitude
+    assert 23.4 == decoded_new_data.speed
+    assert 5 == decoded_new_data.number_of_satellites
+    assert FlightState.NOT_SET == decoded_new_data.flight_state
+    assert True == decoded_new_data.valid
+
+
+def test_position_packet():
+
```
Comment (DarylDohner) [inline tests/format/test_position.py:38]: add docstring

**Review by DarylDohner — CHANGES_REQUESTED** [review]
You forgot to address  this: **You also need to update setup.py to bump the version to 2.0.1**

**Review by DarylDohner — APPROVED** [review]
Approved, tag new version (2.0.1) after merge

**Pull request merged**